### PR TITLE
Refactor create commit

### PR DESCRIPTION
**Pull request by raphaelrobert**

### CODE REVIEW

File: src/tree/test_treemath.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ fn test_tree_hash() {
 
     fn create_identity(id: &[u8], ciphersuite: &Ciphersuite) -> KeyPackageBundle {
```
Comment (franziskuskiefer) [inline src/tree/test_treemath.rs:55]: If this is consuming `ciphersuite`, it shouldn't get a reference.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
I like the API change. But this needs a little more work.
It also conflicts with #29

File: src/tree/test_treemath.rs (hunk 1 above)
Comment (raphaelrobert) [reply src/tree/test_treemath.rs:55]: Let's leave the whole identity stuff for a separate PR

File: src/group/mls_group/create_commit.rs [hunk 2]
```diff
@@ -31,63 +30,49 @@ impl MlsGroup {
...
-            key_package_bundle.private_key,
-            key_package_bundle.key_package,
-        );
-        // Create KeyPackageBundles
-        let mut pending_kpbs = vec![];
-        for kpb in own_key_packages {
-            let (pk, kp) = (kpb.private_key, kpb.key_package);
-            pending_kpbs.push(KeyPackageBundle::from_values(kp, pk));
-        }
+        let mut contains_own_updates = false;
```
[the remark this answers is not in the document]
Comment (raphaelrobert) [reply src/group/mls_group/create_commit.rs:37]: `create_commit_optional_path` in `test_group.rs`

File: src/group/mls_group/api.rs [hunk 3]
```diff
@@ -62,9 +62,7 @@ pub trait Api: Sized {
         &self,
```
[the remark this answers is not in the document]
Comment (raphaelrobert) [reply src/group/mls_group/api.rs:62]: The whole API should actually get better documentation

**Comment by franziskuskiefer** [comment]
I'll leave this open. Let's resolve the two open PRs on Monday together.

File: tests/test_group.rs [hunk 4]
```diff
@@ -1,8 +1,125 @@
...
+        &bob_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        Credential::Basic(bob_credential), // TODO: this consumes the credential!
+        None,
+    );
+    let bob_key_package = bob_key_package_bundle.get_key_package();
+
+    let alice_update_key_package_bundle = KeyPackageBundle::new(
+        &ciphersuite,
+        &alice_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        Credential::Basic(alice_credential),
```
Comment (franziskuskiefer) [inline tests/test_group.rs:42]: Should this be a different credential? (Then we wouldn't have to clone it further up.)

**Review by franziskuskiefer — APPROVED** [review]
two nits

File: tests/test_group.rs (hunk 4 above)
Comment (raphaelrobert) [reply tests/test_group.rs:42]: Not necessarily, a credential can be very long-term.

**Pull request merged**